src/aldente.cpp

```diff
@@ -75,12 +77,30 @@ void Aldente::start_game_loop() {
     // Have window fire off a resize event to update all interested systems.
     window.broadcast_size();
 
+    // Game logic.
+    Phase::curr_phase = PhaseType::BUILD;
+
+    NetworkManager::connect();
+
+    events::joystick_event.connect([&](events::JoystickData d) {
+        if (Phase::curr_phase == PhaseType::BUILD) {
+            if (d.is_button == true && d.input == 0 && d.state == 0) {
+                BuildPhase::is_menu = false;
```

Comment:
You can have the Phase subclasses be static a la [god book](http://gameprogrammingpatterns.com/state.html#static-states)